config/quality/checkstyle/checkstyle.xml

```diff
@@ -1,238 +1,249 @@
+        <module name="RedundantModifier" />
 
         <!-- Checks for Naming Conventions.                  -->
         <!-- See http://checkstyle.sourceforge.net/config_naming.html -->
+        <module name="AbbreviationAsWordInName" />
+        <module name="AbstractClassName" />
+        <module name="CatchParameterName" />
+        <module name="ClassTypeParameterName" />
         <module name="ConstantName" />
+        <module name="InterfaceTypeParameterName" />
         <module name="LocalFinalVariableName" />
-        <module name="FinalLocalVariable">
-            <property name="tokens" value="VARIABLE_DEF,PARAMETER_DEF"/>
-        </module>
         <module name="LocalVariableName" />
         <module name="MemberName" />
-        <module name="MethodName">
-            <property name="severity" value="ignore" />
-        </module>
+        <module name="MethodName" />
+        <module name="MethodTypeParameterName" />
         <module name="PackageName" />
-        <module name="ParameterName">
-            <property name="severity" value="ignore" />
-        </module>
-        <module name="StaticVariableName">
-            <property name="severity" value="ignore" />
-        </module>
+        <module name="ParameterName" />
+        <module name="StaticVariableName" />
         <module name="TypeName" />
 
-        <!-- Checks for imports                              -->
-        <!-- See http://checkstyle.sourceforge.net/config_import.html -->
-        <module name="AvoidStarImport" />
-        <module name="IllegalImport" />
-        <!-- defaults to sun.* packages -->
-        <module name="RedundantImport" />
-        <module name="UnusedImports" />
-
+        <!-- Checks for Naming Conventions.                  -->
+        <!-- See http://checkstyle.sourceforge.net/config_naming.html -->
+        <module name="Regexp" />
 
         <!-- Checks for Size Violations.                    -->
         <!-- See http://checkstyle.sourceforge.net/config_sizes.html -->
+        <module name="AnonInnerLength">
+            <property name="max" value="20" />
+        </module>
+        <module name="ExecutableStatementCount" />
+        <!-- On a modern screen there fit a lot more than 80 characters; 150 is still quite readable -->
         <module name="LineLength">
-            <property name="severity" value="ignore" />
+            <property name="max" value="150" />
         </module>
+        <module name="MethodCount" />
         <module name="MethodLength" />
+        <module name="OuterTypeNumber" />
         <module name="ParameterNumber" />
 
 
         <!-- Checks for whitespace                               -->
         <!-- See http://checkstyle.sourceforge.net/config_whitespace.html -->
+        <module name="EmptyForInitializerPad" />
         <module name="EmptyForIteratorPad" />
-        <module name="GenericWhitespace">
+        <!-- Ignored because it costs a lot of time while it doesn't improve readability -->
```

Comment:
It certainly doesn't cost a lot of time, and I actually think it improves readability, especially in bigger classes...